Runtime/Behaviours/CinemachineDollyCart.cs

```diff
@@ -74,12 +79,33 @@ void LateUpdate()
 
         void SetCartPosition(float distanceAlongPath)
         {
+            //if (m_Path != null)
+            //{
+                //m_Position = m_Path.StandardizeUnit(distanceAlongPath, m_PositionUnits);
+                //transform.position = m_Path.EvaluatePositionAtUnit(m_Position, m_PositionUnits);
+                //transform.rotation = m_Path.EvaluateOrientationAtUnit(m_Position, m_PositionUnits);
+            //}
             if (m_Path != null)
             {
-                m_Position = m_Path.StandardizeUnit(distanceAlongPath, m_PositionUnits);
-                transform.position = m_Path.EvaluatePositionAtUnit(m_Position, m_PositionUnits);
-                transform.rotation = m_Path.EvaluateOrientationAtUnit(m_Position, m_PositionUnits);
+                m_Position = StandardizePathDistance(distanceAlongPath);
+                float3 result = m_Path.EvaluatePosition(m_Position / m_Path.CalculateLength());
```

Comment:
you're calculating the path length twice - once in Standardize, and once here.  Not efficient

Reply:
You are right. I come up with two ways to solve this problem:
1. Pass length as a valuable to both functions
2. Merge `StandardizeUnit` and `EvaluatePositionAtUnit` into one function

Which one do you think is better?

Reply:
Either approach has its advantages, so the decision will depend on the surrounding code.  I suggest that, for ease of conversion from the old system, you build a helper class with an API as close as possible to CinemachinePathBase.  That way, converting legacy code (such as DollyCart) will be trivial.  Don't forget that our customers will have to do something similar if they have custom classes that use paths.

Reply:
What about keep both `StandardizeUnit` and `EvaluatePositionAtUnit` in the helper class? But in `EvaluatePositionAtUnit`, add `StandardizeUnit` before it (copy the code, not calling the function)
Advantages:
1. customers don't need to remember to call `StandardizeUnit` every time they want to use `EvaluatePositionAtUnit`
2. Keep the same api(same function name && same parameters)
Disadvantage:
1. Duplicate code

Reply:
I think you need both.  CinemachinePathBase had both, if I remember correctly.  We should do what it takes to minimize impact on client classes

Reply:
Yea that's what I mean, my changes is add `StandardizeUnit` into `EvaluatePositionAtUnit`, so in `Dolly cart` we only need to call `EvaluatePositionAtUnit`, but in helper class we will have `StandardizeUnit` and `EvaluatePositionAtUnit`.

Reply:
👍